src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -349,10 +349,15 @@ TxQ::canBeHeld(STTx const& tx, OpenView const& view,
             can queue. Mitigates the lost cost of relaying should
             an early one fail or get dropped.
         */
+        auto const tSeq = tx.getSequence();
         canBeHeld = accountIter == byAccount_.end() ||
             replacementIter ||
                 accountIter->second.getTxnCount() <
-                    setup_.maximumTxnPerAccount;
+                    setup_.maximumTxnPerAccount ||
```

Comment:
This construct has now, officially, become impossible to read.

Is this better maybe?
```
/* Limit the number of transactions an individual account
   can queue. Mitigates the lost cost of relaying should
   an early one fail or get dropped.
 */
if (canBeHeld)
{
    if (accountIter != byAccount_.end())
        canBeHeld = false;

    if (!canBeHeld && replacementIter)
        canBeHeld = true;
    
    if (!canBeHeld && 
            accountIter->second.getTxnCount() < setup_.maximumTxnPerAccount)
        canBeHeld =  true;

    // Allow the transaction to get in front of the first queued
    // transaction. This allows recovery of open ledger transactions
    // and stuck transactions.
    if (!canBeHeld &&
            tx.getSequence() < accountIter->second.transactions.begin()->first)
        canBeHeld = true;
}
```

P.S.: I am convinced that the above logic is correct and equivalent to what's already there, but please **double-check** before using this.

Reply:
I updated it differently, but I think you'll like it.